addons/account/models/account_analytic_line.py

```diff
@@ -26,6 +29,55 @@ def _check_company_consistency(self):
         if self._cr.fetchone():
             raise UserError(_("You can't set a different company on your analytic account since there are some journal items linked to it."))
 
+    @api.depends('line_ids')
+    def _compute_invoice_count(self):
+        sale_types = self.env['account.move'].get_sale_types()
+        domain = [
+            ('move_id.state', '=', 'posted'),
+            ('move_id.move_type', 'in', sale_types),
+            ('analytic_account_id', 'in', self.ids)
+        ]
+        groups = self.env['account.move.line'].read_group(domain, ['move_id:count_distinct'], ['analytic_account_id'])
+        moves_count_mapping = dict(map(lambda group: (group['analytic_account_id'][0], group['analytic_account_id_count']), groups))
```

Comment:
You could use a dict comprehension instead. {g['analytic_account_id'][0]: g['analytic_account_id_count'] for g in groups}

Reply:
seems no one likes map reduce ;)

Reply:
Yeah, I didn't point it out because it is fine using the map reduce; but I would have used dict comprehension myself too.
Feels more readable in python.